Docs: SCIM Sailpoint (#56759)
gravitational/teleport: docs/pages/admin-guides/management/guides/scim/sailpoint.mdx

---
title: SCIM Sailpoint Integration
description: How to Configure SCIM Connector in SailPoint to manage Access List membership
---

## Step 1/2: Configure SCIM 2.0 Teleport Connector in SailPoint

To integrate Teleport with SailPoint using SCIM, you need to configure a SCIM connector in SailPoint IdentityNow or SailPoint IdentityIQ.
The exact configuration steps may vary slightly depending on your version of SailPoint, but the general process is as follows:

### Configure SCIM in SailPoint

Go to **Applications > Application Definition > Add New Application**.
Select **SCIM 2.0** as the application type and provide the required configuration details.
Create a new SCIM connector in SailPoint at: Applications > Application Definition > Add New Application.
Select **SCIM 2.0** as the application type and provide the required configuration details:

![Sailpoint Connector Details](../../../../../img/sailpoint/sailpoint-1.png)

> Set the **Base URL** and provided by Teleport under **Configuration -> Settings**.

Author: It this provided in the parent docs: 

Click **Continue** to proceed to the **SCIM Credentials** screen.

- Teleport uses OAuth 2.0 Bearer Tokens for SCIM authentication.
- Copy the **Client ID**, **Client Secret**, and **Base URL** — y**ou'll use them
  when configuring your Identity Provider In next step.**

### Step 3/3. Configure SCIM integration with your Identity Management SCIM provider

- [SailPoint Teleport SCIM connector setup](./SailPoint.mdx)
- In case of Other SCIM providers, please refer to their documentation for SCIM integration.

I have added the SCIM plugin setup as prerequisite to the sailpoint SCIM configuration. 

@ptgott @r0mant  WDUT

My main motivation is to keep the docs page concise and easy to read. While I understand that the SCIM setup can be reused, I feel it might be more clear and maintainable if we decouple the SCIM provider configuration from the Teleport SCIM plugin configuration.

WDYT
Alternatively, do you think it would be better to include all SCIM-related configuration settings within each SCIM provider guide via reusing the scim.mdx common part ?

> Set the **Base URL** and provided by Teleport under **Configuration -> Settings**.

Author: >  Copy the Client ID, Client Secret, and Base URL — you'll use them
when configuring your Identity Provider In next step. 

The step doesn't say that close the integration page. 
This will be the same flow as the one guide in single page.

Set the Authentication Type to OAuth 2.0 with the Grant Type set to **Client Credentials**.
Copy Client ID and Client Secret from the saved Teleport SCIM configuration settings.
Click **Test Connection** to verify that the connection is successful:

![SailPoint SCIM Oauth](../../../../../img/sailpoint/sailspoint-scim-config.png)

### Configure SCIM schema discovery

Under  **Configuration -> Schema**, click **Discover Schema Attributes** on both the **Accounts** and **Groups** tabs to retrieve the schema attributes:
![Account Schema Discovery](../../../../../img/sailpoint/sailpoint-3.png)
![Group Schema Discovery](../../../../../img/sailpoint/sailpoint-4.png)
Go to the **Provisioning Policy** section, and create a **Create Policy** that maps the `userName` SCIM attribute to the user’s email address:
![User Provisioning settings](../../../../../img/sailpoint/sailpoint-5.png)
Save all changes.

#### Configure SCIM group aggregation in SailPoint

SailPoint group aggregation enables the retrieval of SCIM-type Access Lists into SailPoint as Application Group Entitlements.
This allows you import Teleport Access List into SailPoint and membership directly through SailPoint.

> Navigate to **Setup > Tasks -> New Task -> Group Aggregation**.

Author: All these steps are done on the SailPoint side.

![Select Aggregation](../../../../../img/sailpoint/sailpoint-6.png)
Select the **Teleport SCIM Connector**, then click **Save and Execute** to run the aggregation task.
![Configure Aggregation](../../../../../img/sailpoint/sailpoint-7.png)
If the aggregation completes successfully,
you should see the imported Access Lists `type: "scim"` from Teleport in SailPoint under: **Applications > Entitlement Catalog**

![Aggregation Result](../../../../../img/sailpoint/sailpoint-8.png)

## Step 2/2: Submit Access Requests to SailPoint Group Entitlement (Optional)

Go to Manage > Manage User Access > Manage User Access.
Submit an Access Request for a mapped Access List (as represented by a group
entitlement in SailPoint).
![Access request](../../../../../img/sailpoint/sailpoint-9.png)

> Once the request is approved, the user will be added to the appropriate Access List in Teleport.

Author: This is already described in https://github.com/gravitational/teleport/pull/56759/files#diff-5b05c1e5077ffbf44f4446b4ef4266498313761e26308842a28ee642ed6460a6R18

---
title: SCIM SailPoint Integration
description: How to Configure SCIM Connector in SailPoint to manage Access List membership
---

> SailPoint provides the SCIM Identity Management Connector that allows to manage Teleport Access List membership
> through SailPoint IdentityNow or SailPoint IdentityIQ.

Reviewer: ```suggestion
SailPoint provides a SCIM identity management connector that allows you to manage Teleport Access List membership
through SailPoint IdentityNow or SailPoint IdentityIQ.
```
Grammar fixes

## Prerequisites
- Teleport SCIM plugin setup: [SCIM Plugin Installation](./scim.mdx)

## Step 1/2: Configure a SCIM 2.0 Teleport connector in SailPoint

To integrate Teleport with SailPoint using SCIM, you need to configure a SCIM connector in SailPoint IdentityNow or SailPoint IdentityIQ.
The exact configuration steps may vary slightly depending on your version of SailPoint, but the general process is as follows.

### Configure SCIM in SailPoint

Create a new SCIM connector in SailPoint at: Applications > Application Definition > Add New Application.
Select **SCIM 2.0** as the application type and provide the required configuration details:

![SailPoint Connector Details](../../../../../img/sailpoint/sailpoint-1.png)

> Navigate to the Teleport SCIM Integration page and copy the OAuth 2.0 details and the SCIM Base URL.

Reviewer: ```suggestion
Navigate to the Teleport SCIM Integration page and copy the OAuth 2.0 details and the SCIM base URL.
```
Not a proper noun

<Admonition type="info">
    If the Teleport SCIM integration has not been set up, follow the [SCIM Plugin Installation](./scim.mdx)  guide before proceeding.
</Admonition>

In the SailPoint SCIM application configuration view, populate the configuration settings using the values obtained from the Teleport SCIM integration:

> * SCIM Base URL: Paste the SCIM Base URL copied from SCIM Teleport Integration page.
> * Authentication Type: OAuth 2.0
> * Token URL: Paste the Token UR copied from SCIM Teleport Integration page.
> * Grant Type: **Client Credentials**
> * **Client ID**: The Client ID copied from SCIM Teleport Integration pag

Reviewer 2: Nit: Just for a bit more consistency
```suggestion
* SCIM Base URL: Paste the SCIM Base URL copied from SCIM Teleport Integration page
* Authentication Type: Pick "OAuth 2.0"
* Token URL: Paste the Token UR copied from SCIM Teleport Integration page
* Grant Type: Pick "Client Credentials"
* Client ID: The Client ID copied from SCIM Teleport Integration page
```

Click **Test Connection** to verify that the connection is successful:

![SailPoint SCIM Oauth](../../../../../img/sailpoint/sailspoint-scim-config.png)

### Configure SCIM schema discovery

Under  **Configuration -> Schema**, click **Discover Schema Attributes** on both the **Accounts** and **Groups** tabs to retrieve the schema attributes:
![Account Schema Discovery](../../../../../img/sailpoint/sailpoint-3.png)
![Group Schema Discovery](../../../../../img/sailpoint/sailpoint-4.png)
Go to the **Provisioning Policy** section, and create a **Create Policy** that maps the `userName` SCIM attribute to the user’s email address:
![User Provisioning settings](../../../../../img/sailpoint/sailpoint-5.png)
Save all changes.

#### Configure SCIM group aggregation in SailPoint

SailPoint group aggregation enables the retrieval of SCIM-type Access Lists into SailPoint as Application Group Entitlements.

> This allows you import Teleport Access List into SailPoint and membership directly through SailPoint.

Reviewer 2: ```suggestion
This allows you to import Teleport Access List into SailPoint and manage membership directly through SailPoint with the changes being reflected in Teleport Access Lists.
```

> This allows you to import Teleport Access List into SailPoint and manage membership directly through SailPoint with the changes being reflected in Teleport Access Lists.

Reviewer: ```suggestion
This allows you to import Teleport Access Lists into SailPoint and manage membership directly through SailPoint with the changes being reflected in Teleport Access Lists.
```
Grammar
